sdjournal/journal.go

```diff
@@ -260,6 +262,61 @@ func (j *Journal) GetDataValue(field string) (string, error) {
 	return strings.SplitN(val, "=", 2)[1], nil
 }
 
+// GetDataMap returns all the Key-Value pairs of data for the current journal
+// entry. Mimics the behavior of output_export in logs-show.c
+func (j *Journal) GetDataMap() (map[string]string, error) {
+	var r C.int
+	var d unsafe.Pointer
+	var l C.size_t
+
+	resultMap := make(map[string]string)
+
+	j.mu.Lock()
+	defer j.mu.Unlock()
+
+	var realtimeUsec C.uint64_t
+	r = C.sd_journal_get_realtime_usec(j.cjournal, &realtimeUsec)
+	if r < 0 {
+		return nil, fmt.Errorf("Failed to get realtime timestamp: %d", r)
+	}
+	resultMap["__REALTIME_TIMESTAMP"] = fmt.Sprintf("%d", realtimeUsec)
```

Comment:
can you split these into consts as we have for https://github.com/coreos/go-systemd/pull/136/files#diff-167dc1dd14a090cd485c7dfbb2cfa9b3L42 ?